client/core/core.go

```diff
@@ -76,6 +76,48 @@ type dexConnection struct {
 
 	regConfMtx  sync.RWMutex
 	regConfirms uint32
+
+	pendingSuspendsMtx sync.Mutex
+	pendingSuspends    map[string]*time.Timer
```

Comment:
Yes it does make sense but  `marketMtx` is for `marketMap` in `dexConnection`. Looks like `Market` should have its own mutex for `suspend` and  `pendingSuspend`.

Reply:
Oh right. It only protects the map. This is trickier then, since `Market` is exported. Might need an internal `xcMarket` type with a mutex and a method for generating the exported `Market`.

Reply:
Oops just reading this, will update the commit just pushed.

Reply:
I think the simpler option would be to unexport `Suspended` and provide a getter/setter which will utilize the mutex. Thoughts?

Reply:
Should work.